pallets/settlement/src/lib.rs

```diff
@@ -1522,57 +1524,109 @@ impl<T: Trait> Module<T> {
                 receipt.metadata.clone(),
             ));
         }
-
-        // Execute instruction if conditions are met.
-        let execute_instruction_weight = Self::is_instruction_executed(
-            affirms_pending,
-            instruction_details.settlement_type,
-            instruction_id,
-        );
-
-        execute_instruction_weight.map(|info| {
-            info.actual_weight
-                .map(|w| {
-                    w.saturating_add(weight_for::weight_for_affirmation_with_receipts::<T>(
-                        u32::try_from(receipt_details.len()).unwrap_or_default(),
-                    ))
-                })
-                .into()
-        })
+        Ok(())
     }
 
     pub fn base_affirm_instruction(
         origin: <T as frame_system::Trait>::Origin,
         instruction_id: u64,
         portfolios: Vec<PortfolioId>,
-    ) -> DispatchResultWithPostInfo {
+    ) -> DispatchResult {
         let (did, secondary_key) =
             Self::ensure_origin_perm_and_instruction_validity(origin, instruction_id)?;
         let portfolios_set = portfolios.into_iter().collect::<BTreeSet<_>>();
 
         // Provide affirmation to the instruction
-        Self::unsafe_affirm_instruction(
-            did,
+        Self::unsafe_affirm_instruction(did, instruction_id, portfolios_set, secondary_key.as_ref())
+    }
+
+    // It affirms the instruction and may schedule the instruction
+    // depends on the settlement type.
+    pub fn affirm_with_receipts_and_maybe_schedule_instruction(
+        origin: <T as frame_system::Trait>::Origin,
+        instruction_id: u64,
+        receipt_details: Vec<ReceiptDetails<T::AccountId, T::OffChainSignature>>,
+        portfolios: Vec<PortfolioId>,
+    ) -> DispatchResult {
+        Self::base_affirm_with_receipts(origin, instruction_id, receipt_details, portfolios)?;
+        // Schedule instruction to be execute in the next block (expected) if conditions are met.
+        Self::maybe_schedule_instruction(
+            Self::instruction_affirms_pending(instruction_id),
+            Self::instruction_details(instruction_id).settlement_type,
             instruction_id,
-            portfolios_set,
-            secondary_key.as_ref(),
-        )?;
+        )
+    }
 
-        // Execute the instruction if conditions are met
-        let affirms_pending = Self::instruction_affirms_pending(instruction_id);
-        let weight_for_instruction_execution = Self::is_instruction_executed(
-            affirms_pending,
+    /// It is used for general purpose settlement where instruction get affirmation &
+    /// may schedule for the next block execution or it is assumed that it is already scheduled.
```

Comment:
Please rewrite this doc comment in the style as I have below.
It's standard to use imperative/active present tense in rust docs.
(Applies generally in PR.)